### PR TITLE
[dif] Add glen limit to csrng commands.

### PR DESCRIPTION
**Pull request by moidx**
Return `kDifOutOfRange` errors if the glen command is equivalent to a value greater than 0x800 128 bit blocks.

This is part of #18334.

### CODE REVIEW

File: sw/device/tests/usbdev_stream_test.c [hunk 1]
```diff
@@ -44,7 +44,7 @@
 #define TRANSFER_BYTES_VERILATOR 0x2400U
 
 // This is about the amount that we can transfer within a 1 hour 'eternal' test
-//#define TRANSFER_BYTES_LONG (0xD0U << 20)
+// #define TRANSFER_BYTES_LONG (0xD0U << 20)
```
Comment (vogelpi) [inline sw/device/tests/usbdev_stream_test.c:47]: This seems to be unrelated, is it added on purpose?

File: sw/device/lib/dif/dif_csrng.h [hunk 2]
```diff
@@ -398,6 +398,8 @@ dif_result_t dif_csrng_update(const dif_csrng_t *csrng,
  *
  * @param csrng A CSRNG handle.
  * @param len Number of uint32_t words to generate.
+ * @return The result of the operation. KDifOutOfRange if the `len` parameter is
```
Comment (vogelpi) [inline sw/device/lib/dif/dif_csrng.h:401]: Nit: remove "is"

File: sw/device/lib/dif/dif_edn.h [hunk 3]
```diff
@@ -393,6 +393,8 @@ dif_result_t dif_edn_update(const dif_edn_t *edn,
  *
  * @param edn An EDN handle.
  * @param len Number of uint32_t words to generate.
+ * @return The result of the operation. KDifOutOfRange if the `len` parameter is
```
Comment (vogelpi) [inline sw/device/lib/dif/dif_edn.h:396]: Nit: remove "is"

**Review by vogelpi — APPROVED** [review]
Thanks @moidx! This looks good.

File: sw/device/tests/usbdev_stream_test.c (hunk 1 above)
Comment (moidx) [reply sw/device/tests/usbdev_stream_test.c:47]: It was probably triggered by `clang_format`. I will revert it :)

File: sw/device/lib/dif/dif_csrng.h (hunk 2 above)
Comment (moidx) [reply sw/device/lib/dif/dif_csrng.h]: Done.

File: sw/device/lib/dif/dif_edn.h (hunk 3 above)
Comment (moidx) [reply sw/device/lib/dif/dif_edn.h]: Done.

File: sw/device/lib/crypto/drivers/entropy.c [hunk 4]
```diff
@@ -230,6 +230,16 @@ OT_WARN_UNUSED_RESULT
...
                                    entropy_csrng_cmd_t cmd,
                                    bool check_completion) {
+  enum {
+    // This is to maintain full compliance with NIST SP 800-90A, which requires
+    // the max generate output to be constrained to gen < 2 ^ 12 bits or 0x800
+    // 128-bit blocks.
+    kMaxGenerateSizeIn128BitBlocks = 0x800,
+  };
+  if (cmd.generate_len > kMaxGenerateSizeIn128BitBlocks) {
+    return OUT_OF_RANGE();
```
Comment (jadephilipoom) [inline sw/device/lib/crypto/drivers/entropy.c:240]: nit: sorry for the post-merge comment, but it might be good to use a cryptolib status macro here so it's clear which top-level error code it will become:
```suggestion
return OTCRYPTO_BAD_ARGS;
```

As written, this will get translated into an "internal error" in the top-level API, which is the default for most error codes that aren't specifically mapped to one of the top-level status codes.
https://github.com/lowRISC/opentitan/blob/39f9a9e57286f0d2a72c624bb19b9bd3a4c3bdf0/sw/device/lib/crypto/impl/status.c#L43-L44
